libdnf/rpm/solv_query.cpp

```diff
@@ -1794,9 +1794,13 @@ SolvQuery & SolvQuery::ifilter_installed() {
         Solvable * solvable = solv::get_solvable(pool, candidate_id);
         if (solvable->repo == installed_repo) {
             filter_result.add_unsafe(candidate_id);
+            continue;
```

Comment:
`for (PackageId candidate_id : p_impl->query_result) {`
It would be also nice if we can start to iterate from `installed_repo->start`

Reply:
Yes, that would be nice. But in my commit message is explanation:
Theoretically, the iteration can start from "installed_repo->start",
**but we can't set iterator to this position. The iterator does not have
a setter only an operator ++. -> Iteration must start from the beginning
of the pool.** There is usually no slowdown because the installed repository
is usually the first repository in the pool.

But I have already solved the problem. I will prepare a commit.

Reply:
I added `jump()` method into `rpm::solv::SolvMapIterator` in PR https://github.com/rpm-software-management/libdnf/pull/1092 .
And the current version of the optimization uses it to start the iteration from `installed_repo-> start`.